src/pal/src/file/file.cpp

```diff
@@ -1716,8 +1776,12 @@ GetFileAttributesW(
         goto done;
     }
     
-    size = WideCharToMultiByte( CP_ACP, 0, lpFileName, -1, filename, MAX_LONGPATH,
+    length = (PAL_wcslen(lpFileName)+1) * MaxWCharToAcpLengthFactor;
```

Comment:
Yes i did think about adding an helper function, but i didnt find a good place to add that, since this pattern is used all over the repo (debug, zap, file, loader, etc). Where would be a good place to add this? I will not make this change for this PR, since it's already in, and it's not going to change the behavior. But if i do encounter this pattern in future will add it then if there is a need.